File: GameLayers/drawCardsLayer.js
import { config } from '../config.js';

export class RebuildInitialHandDrawCardsLayer {
    constructor(hand){
        this.hand = hand
    }

    execute(world){
        world.heroHand = [...this.hand]
    }
}

export class DrawFromDeckDrawCardsLayer {
    execute(world){
        while(world.heroHand.length < config.HAND_SIZE)
            world.heroHand.push(world.heroDeck.pop())
    }
}

File: python/GameLayers/drawCardsLayer.py
import config

class RebuildInitialHandDrawCardsLayer:
    def __init__(self, hand):
        self.hand = hand

    def execute(self, world):
        world.heroHand = self.hand

class DrawFromDeckDrawCardsLayer:
    def execute(self, world):
        while(len(world.heroHand) < config.HAND_SIZE):
            world.heroHand.append(world.heroDeck.pop())


